synapseclient/api/evaluation_services.py

```diff
@@ -0,0 +1,401 @@
+"""
+This module is responsible for exposing the services defined at:
+<https://rest-docs.synapse.org/rest/index.html#org.sagebionetworks.repo.web.controller.EvaluationController>
+"""
+
+import json
+from typing import TYPE_CHECKING, List, Optional
+
+if TYPE_CHECKING:
+    from synapseclient import Synapse
+
+
+async def create_or_update_evaluation(
+    request_body: dict,
+    *,
+    synapse_client: Optional["Synapse"] = None,
+) -> dict:
+    """
+    Create or update an Evaluation on Synapse.
+
+    <https://rest-docs.synapse.org/rest/POST/evaluation.html>
+    <https://rest-docs.synapse.org/rest/PUT/evaluation/evalId.html>
+
+    Arguments:
+        request_body: A dictionary containing the evaluation data
+        synapse_client: If not passed in and caching was not disabled by `Synapse.allow_client_caching(False)` this will use the last created
+                        instance from the Synapse class constructor.
+
+    Returns:
+        The Evaluation response object as a raw JSON dict.
+
+    Raises:
+        SynapseHTTPError: If the service rejects the request or an HTTP error occurs.
+    """
+    import logging
+
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    logger = client.logger if client else logging.getLogger(__name__)
+
+    if not request_body.get("id"):
+        uri = "/evaluation"
+        response = await client.rest_post_async(uri, body=json.dumps(request_body))
+
+        logger.info(
+            f"Evaluation '{request_body.get('name')}' has been created with ID: {response.get('id')}"
+        )
+
+    elif request_body.get("id"):
```

Comment:
nit: Could this just be an else statement? `id` either is or isn't in the `request_body` right?